### PR TITLE
add parameters to embedded-linux-pipeline: accessLoggingBucket, artif…

### PR DESCRIPTION
**Pull request by thomas-roos**
…actBucket, outputBucket

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source-repo/poky-ami/build.buildspec.yml [hunk 1]
```diff
@@ -63,8 +63,10 @@ phases:
       - find /downloads -atime +30 -type f -delete
       # only delete empty dirs
       - find /downloads -atime +30 -type d -empty -delete
+
 artifacts:
-  discard-paths: true
+  s3-prefix: poky-ami/
```
Comment (alinadima) [inline source-repo/poky-ami/build.buildspec.yml:68]: Did you evaluate if there is another possibility to build an S3 path outside of the build spec and pass it into code build? Maybe you can use the objectKey property of the S3DeployActionProps? 

1. What happens if the customer wants to have multiple buckets? We said: default - one bucket and prefix, but if desired, customer could build a bucket per pipeline too. 

2. I have concerns about the implications of having the prefix in the build spec. The build spec is owned by the customer, but the library owns the infrastructure. If we do this in the build spec, we need to document it, and we would need to create a basic proposed build spec template for example to ensure mandatory params are in. What happens if the customer decides to remove the prefix in the build spec? Will data be overwritten?

Comment (thomas-roos) [reply source-repo/poky-ami/build.buildspec.yml:68]: 1. he needs to change the buildspec AND app.ts to write to a different bucket
Agree on 2. yes - data will be overwritten.

Still struggling to upload to a s3 bucket subdir - seems that s3-prefix is not the right thing.

Comment (thomas-roos) [reply source-repo/poky-ami/build.buildspec.yml:68]: S3DeployActionProps works

File: lib/build-image-pipeline.ts [hunk 2]
```diff
@@ -29,6 +29,10 @@ export interface BuildImagePipelineProps extends cdk.StackProps {
   readonly dataBucket: s3.IBucket;
   /** The ECR Repository to push to. */
   readonly repository: IRepository;
+  /** Access logging bucket to use */
+  accessLoggingBucket?: s3.Bucket;
```
Comment (nateglims) [inline lib/build-image-pipeline.ts:33]: We should also expose the prefix for sharing a bucket: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_s3.Bucket.html#serveraccesslogsprefix

File: lib/embedded-linux-pipeline.ts [hunk 3]
```diff
@@ -236,11 +262,22 @@ export class EmbeddedLinuxPipelineStack extends cdk.Stack {
...
 
-    const artifactAction = new codepipeline_actions.S3DeployAction({
-      actionName: 'Artifact',
-      input: buildOutput,
-      bucket: outputBucket,
-    });
+    let artifactAction: codepipeline_actions.S3DeployAction;
+
+    if (props.subDirectoryName){
+      artifactAction = new codepipeline_actions.S3DeployAction({
```
Comment (nateglims) [inline lib/embedded-linux-pipeline.ts:268]: Only some things get into the final bucket with this action. There is also some things in the build spec files.

Comment (thomas-roos) [reply lib/embedded-linux-pipeline.ts:268]: What do you mean by "some things" ?
By setting the subDirectoryName all uploaded artifacts go into a sub dir...

File: lib/build-image-pipeline.ts (hunk 2 above)
Comment (thomas-roos) [reply lib/build-image-pipeline.ts:33]: agree

Comment (thomas-roos) [reply lib/build-image-pipeline.ts]: done

**Pull request merged**